### PR TITLE
Deprecate std::net::lookup_host

### PR DESCRIPTION
**Pull request by sfackler**
We intended to do this quite a while ago but it snuck through.

r? @alexcrichton

### CODE REVIEW

File: src/libstd/net/mod.rs [hunk 1]
```diff
@@ -134,12 +134,15 @@ fn each_addr<A: ToSocketAddrs, F, T>(addr: A, mut f: F) -> io::Result<T>
                                               iterator and returning socket \
                                               addresses",
            issue = "27705")]
+#[rustc_deprecated(since = "1.26", reason = "Use the ToSocketAddrs trait instead")]
```
Comment (kennytm) [inline src/libstd/net/mod.rs]: The current nightly version is still 1.25...?

**Comment by alexcrichton** [comment]
r=me with the typo @kennytm pointed out

**Pull request merged**